shared/js/intl_helper.js

```diff
@@ -0,0 +1,170 @@
+(function(global) {
+  'use strict';
+
+  /*************************
+
+  IntlHelper.define('shortDate', 'datetime', {
+    year: 'numeric',
+    month: 'long',
+  });
+
+  function setValue() {
+    var formatter = IntlHelper.get('shortDate');
+    console.log(formatter.format(new Date()));
+  }
+
+  setValue();
+
+  IntlHelper.observe('shortDate', setValue);
+  
+   ****************************/
+  var objectCache;
```

Comment:
As you know, 1) I'm not a fan of calling things "objects" and 2) I'm bad at naming.  But perhaps this could called `helperCache`?  Compare:

    var obj = objectCache.get(name);
    obj.object = ...

with:

    var helper = helperCache.get(name);
    helper.intlObject = ...